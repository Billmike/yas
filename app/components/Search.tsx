import React from 'react';
import { View, StyleSheet, TextInput, TextInputProps } from 'react-native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';

interface ISearch extends TextInputProps { }

const styles = StyleSheet.create({
  searchView: {
    flexDirection: 'row',
    borderColor: 'black',
    borderWidth: 1,
    borderRadius: 8,
    height: 40,
    alignItems: 'center',
    paddingHorizontal: 15,
    marginTop: 20,
  },
  input: {
    flex: 1,
    paddingLeft: 10,
  },
});

const Search = ({ onChangeText, value }: ISearch) => {
  return (
    <View style={styles.searchView}>
      <MaterialIcons testID="search-icon" name="search" size={25} />
      <TextInput value={value} testID="search" placeholder="Find a character" style={styles.input} onChangeText={onChangeText} />
    </View>
  );
};

export default Search;
