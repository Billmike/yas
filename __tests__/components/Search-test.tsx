import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import Search from '../../app/components/Search';

test('The Search Input renders correctly', () => {
  const { toJSON } = render(<Search />);
  expect(toJSON()).toMatchSnapshot();
});

test('Input should have a search icon', async () => {
  const { getByTestId } = render(<Search />);
  const searchIcon = getByTestId('search-icon');
  expect(searchIcon).toBeTruthy();
});

test('The search input is updated as the user types', async () => {
  const mockOnChangeText = jest.fn();
  const { getByTestId } = render(<Search onChangeText={mockOnChangeText} />);
  const input = getByTestId('search');

  fireEvent.changeText(input, 'Rick Sanchez');
  expect(mockOnChangeText).toHaveBeenCalledWith('Rick Sanchez');
});
