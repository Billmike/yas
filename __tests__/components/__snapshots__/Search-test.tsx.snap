// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Search Input renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "black",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 40,
      "marginTop": 20,
      "paddingHorizontal": 15,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 25,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    testID="search-icon"
  >
    î¢¶
  </Text>
  <TextInput
    allowFontScaling={true}
    placeholder="Find a character"
    rejectResponderTermination={true}
    style={
      Object {
        "flex": 1,
        "paddingLeft": 10,
      }
    }
    testID="search"
    underlineColorAndroid="transparent"
  />
</View>
`;
